Public/Src/Engine/Processes/ReportedFileAccess.cs

```diff
@@ -622,6 +623,69 @@ public bool TryParseAbsolutePath(PipExecutionContext context, Process pip, out A
                 enumeratePattern);
         }
 
+        /// <nodoc />
+        public void Serialize(
+            BuildXLWriter writer,
+            Dictionary<ReportedProcess, int> processMap,
+            Action<BuildXLWriter, AbsolutePath> writePath)
+        {
+            writer.Write((byte)Operation);
+
+            if (processMap != null && processMap.TryGetValue(Process, out int index))
+            {
+                writer.WriteCompact(index);
+            }
+            else
+            {
+                Process.Serialize(writer);
+            }
+
+            writer.WriteCompact((int)RequestedAccess);
+            writer.WriteCompact((int)Status);
+            writer.Write(ExplicitlyReported);
+            writer.Write(Error);
+            writer.Write(Usn.Value);
+            writer.Write((uint)DesiredAccess);
+            writer.Write((uint)ShareMode);
+            writer.Write((uint)CreationDisposition);
+            writer.Write((uint)FlagsAndAttributes);
+
+            if (writePath != null)
+            {
+                writePath.Invoke(writer, ManifestPath);
+            }
+            else
+            {
+                writer.Write(ManifestPath);
+            }
+
+            writer.WriteNullableString(Path);
+            writer.WriteNullableString(EnumeratePattern);
+        }
+
+        /// <nodoc />
+        public static ReportedFileAccess Deserialize(
+            BuildXLReader reader, 
+            IReadOnlyList<ReportedProcess> processes, 
+            Func<BuildXLReader, AbsolutePath> readPath)
+        {
+            return new ReportedFileAccess(
+                operation: (ReportedFileOperation)reader.ReadByte(),
+                process: processes != null ? processes[reader.ReadInt32Compact()] : ReportedProcess.Deserialize(reader),
+                requestedAccess: (RequestedAccess)reader.ReadInt32Compact(),
+                status: (FileAccessStatus)reader.ReadInt32Compact(),
+                explicitlyReported: reader.ReadBoolean(),
+                error: reader.ReadUInt32(),
+                usn: new Usn(reader.ReadUInt64()),

```

Comment:
> usn: new Usn(reader.ReadUInt64() [](start = 15, length = 33)

Do USN's actually translate well accross VM's?

Reply:
In general, they don't. But note that we're going to map host volume in the VM, and thus the USN's can still be used.

---
In reply to: [276816840](https://github.com/Microsoft/BuildXL/pull/159#discussion_r276816840) [](ancestors = 276816840)

Reply:
Please add a comment and assert to the effect that it happens to be a magic VM by cloudbuild that happens to run on this box. 
There is nothing that precludes someone in the future to change the VM and put it on single machine inthe machine pool to lock down futher and how we are conflating usn's. I pitty the person that will have to debug it then.

---
In reply to: [277070365](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277070365) [](ancestors = 277070365,276816840)

Reply:
I'll add it in the next iteration.

---
In reply to: [277084423](https://github.com/Microsoft/BuildXL/pull/159#discussion_r277084423) [](ancestors = 277084423,277070365,276816840)